java/src/main/java/com/powsybl/dataframe/network/modifications/VoltageLevelTopologyCreation.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.dataframe.network.modifications;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.ReporterModel;
+import com.powsybl.dataframe.SeriesMetadata;
+import com.powsybl.dataframe.update.UpdatingDataframe;
+import com.powsybl.iidm.modification.topology.CreateVoltageLevelTopologyBuilder;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.SwitchKind;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.powsybl.dataframe.network.adders.SeriesUtils.applyIfPresent;
+
+/**
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+public class VoltageLevelTopologyCreation implements NetworkModification {
+
+    private static final List<SeriesMetadata> METADATA = List.of(
+            SeriesMetadata.stringIndex("id"),
+            SeriesMetadata.ints("low_busbar_index"),
+            SeriesMetadata.ints("busbar_count"),
+            SeriesMetadata.ints("low_section_index"),
+            SeriesMetadata.ints("section_count"),
+            SeriesMetadata.strings("busbar_section_prefix_id"),
+            SeriesMetadata.strings("switch_prefix_id"),
+            SeriesMetadata.strings("switch_kinds")
+    );
+
+    @Override
+    public List<SeriesMetadata> getMetadata() {
+        return METADATA;
+    }
+
+    private CreateVoltageLevelTopologyBuilder createBuilder(UpdatingDataframe dataframe, int row) {
+        CreateVoltageLevelTopologyBuilder builder = new CreateVoltageLevelTopologyBuilder();
+        String switchKindStr = dataframe.getStringValue("switch_kinds", row).isEmpty() ||
```

Comment:
we should avoid to call `getStringValue` 3 times, only one is necessary (there can be a small performed for each call)